models/classes/Lists/Business/Service/RemoteSourceJsonPathParser.php

```diff
@@ -14,48 +15,76 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
- *
+ * Copyright (c) 2020-2021 (original work) Open Assessment Technologies SA;
  */
 
 declare(strict_types=1);
 
 namespace oat\tao\model\Lists\Business\Service;
 
+use RuntimeException;
 use Flow\JSONPath\JSONPath;
 use Flow\JSONPath\JSONPathException;
 use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\Context\ContextInterface;
 use oat\tao\model\Lists\Business\Domain\Value;
-use RuntimeException;
+use oat\tao\model\Lists\Business\Domain\RemoteSourceContext;
 
 class RemoteSourceJsonPathParser extends ConfigurableService implements RemoteSourceParserInterface
 {
     /**
-     * @inheritDoc
+     * @deprecated use $this->iterateContext()
      */
     public function iterate(array $json, string $uriRule, string $labelRule): iterable
     {
-        $jsonPath = new JSONPath($json);
+        yield from $this->iterateContext(
+            new RemoteSourceContext([
+                RemoteSourceContext::PARAM_JSON => $json,
+                RemoteSourceContext::PARAM_URI_PATH => $uriRule,
+                RemoteSourceContext::PARAM_LABEL_PATH => $labelRule,
+            ])
+        );
+    }
+
+    public function iterateContext(ContextInterface $context): iterable
```

Comment:
https://github.com/oat-sa/tao-core/pull/3045/files#r684744084

```suggestion
    public function iterateByContext(ContextInterface $context): iterable
```